app/Controller/SocialController.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+class SocialController extends AppController
+{
+    private $social_default = [
+        ['title' => 'Discord', 'extra' => 'fab fa-discord', 'color' => '#7289da'],
+        ['title' => 'Twitter', 'extra' => 'fab fa-twitter', 'color' => '#00acee'],
+        ['title' => 'Youtube', 'extra' => 'fab fa-youtube', 'color' => '#c4302b'],
+        ['title' => 'FaceBook', 'extra' => 'fab fa-facebook', 'color' => '#3b5998']
+    ];
+
+    function admin_index() {
+        if (!$this->isConnected || !$this->Permissions->can('MANAGE_SOCIAL'))
+            throw new ForbiddenException();
+
+        $this->set('title_for_layout', $this->Lang->get('SOCIAL__HOME'));
+        $this->layout = 'admin';
+
+        $this->loadModel('SocialButton');
+        $this->set('social_buttons', $this->SocialButton->find('all', ['order' => 'order']));
+    }
+
+    public function admin_save_ajax() {
+        $this->autoRender = false;
+        $this->response->type('json');
+        if ($this->isConnected AND $this->Permissions->can('MANAGE_SOCIAL')) {
+            if ($this->request->is('post')) {
+                if (!empty($this->request->data)) {
+                    $data = $this->request->data['social_button_order'];

```

Comment:
Remplace par $this->request->data['xss']['social_button_order'];

Reply:
Merci, je me suis arraché les cheveux pendants des heures pour uniquement ça 😅, question on va devoir faire ça pour chaque request ? et pourquoi d'autres request fonctionnes ?

Reply:
Pour les requêtes où l'anti xss pose problème, ça psoe pb ici car y'a des & et des = ducoup au moment du traitement ça pose pb